Lombiq.Tests.UI.Samples/Helpers/SetupHelpers.cs

```diff
@@ -65,5 +68,27 @@ public static async Task<Uri> RunAutoSetupAsync(UITestContext context)
         return context.GetCurrentUri();
     }
 
-    private static void AssertSetupSuccessful(UITestContext context) => context.Exists(By.Id("navbar"));
+    private static void AssertSetupSuccessful(UITestContext context)
+    {
+        try
+        {
+            context.Exists(By.Id("navbar"));
+        }
+        catch (NoSuchElementException)
+        {
+            var validationErrors = context.GetAll(By.ClassName("field-validation-error"));
+
+            if (!validationErrors.Any()) throw;
+
+            var isSingle = validationErrors.Count == 1;
+            var plural = isSingle ? string.Empty : "s";
+            var errors = isSingle
+                ? " " + validationErrors.Single().Text.Trim()
+                : validationErrors
+                    .Select(element => "\n- " + element.Text.Trim())
+                    .Join(string.Empty);
+
+            throw new AssertionException($"Setup has failed with the following validation error{plural}:{errors}");
```

Comment:
How slick 😁 And very nerdy, too 😉 

Would you despise having a newline and a hyphen when there's only a single error, or were you just having a good time 🥳? Because this could - as you most certainly know - become:

```suggestion
            var errors = "\n- " + validationErrors.Select(element => element.Text.Trim()).Join("\n- ");
            throw new AssertionException($"Setup has failed with the following validation errors:{errors}");
```

Or:

```suggestion
            var errors = new[] { string.Empty }.Concat(validationErrors).Select(element => element.Text.Trim()).Join("\n- ");
            throw new AssertionException($"Setup has failed with the following validation errors:{errors}");
```

Reply:
I didn't want to say "errors" when it only returns one error. Everything else is just flourish.